integrations/gcp/gcp_core/search/searches.py

```diff
@@ -0,0 +1,186 @@
+from typing import Any
+
+from google.api_core.exceptions import NotFound, PermissionDenied
+from google.cloud.asset_v1 import (
+    AssetServiceAsyncClient,
+)
+from google.cloud.asset_v1.services.asset_service import pagers
+from google.cloud.resourcemanager_v3 import (
+    FoldersAsyncClient,
+    OrganizationsAsyncClient,
+    ProjectsAsyncClient,
+)
+from google.pubsub_v1.services.publisher import PublisherAsyncClient
+from loguru import logger
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE, RAW_ITEM
+from port_ocean.utils.cache import cache_iterator_result
+
+from gcp_core.search.utils import (
+    EXTRA_PROJECT_FIELD,
+    AssetTypesWithSpecialHandling,
+    parse_protobuf_message,
+    parse_protobuf_messages,
+)
+
+
+async def search_all_resources(
+    project: dict[str, Any], asset_type: str
```

Comment:
i dont understand the project type

Reply:
This is the project's data, down below i've extracted the project's name and used it, changing to project_data